### PR TITLE
[RLlib; Offline RL] 2. Multiple optimizations for streaming data.

### PR DESCRIPTION
**Pull request by simonsays1980**
## Why are these changes needed?

Streaming data directly from cloud storage showed still low performance. This PR comes with multiple optimizations to train Offline RL agents in a truly streaming way, i.e. starting training after the first chunks of data are read and preprocessed. The following changes were made:
- Removing `schema` calls.
- Removing customizations of data block numbers.
- Removing default filesystems.
- Removing `locality_hints` from `OfflinePreLearner` as this is not used anymore.
- Removing local shuffling of data.
- Removing corresponding customizations from all our `tuned_examples` and tests.
- Changing the `RLUnplugged` example to 
     - Read converted Parquet data to demonstrate the powerful streaming capabilities of Ray Data in our Offline RL API
     - Providing a `SPREAD` scheduling strategy when using multiple learners on a multi-node cluster.
     - Removing all customizations that slow down data processing.
     - Fixing multiple bugs and improving performance in the custom `ConnectorV2` to read stacked and encoded Atari frames

## Related issue number

Related to #49194 

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [x] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/offline/offline_prelearner.py [hunk 1]
```diff
@@ -86,7 +86,6 @@ def __init__(
         self,
         config: "AlgorithmConfig",
         learner: Union[Learner, list[ActorHandle]],
-        locality_hints: Optional[List[str]] = None,
```
Comment (sven1977) [inline rllib/offline/offline_prelearner.py:89]: Let's change the signature to:
```
OfflinePreLearner.__init__(self, *, config, ...) # <- force named keyword args
```

File: rllib/tuned_examples/bc/cartpole_bc.py [hunk 2]
```diff
@@ -52,21 +52,13 @@
     # as remote learners.
     .offline_data(
         input_=[data_path.as_posix()],
-        # Define the number of reading blocks, these should be larger than 1
```
Comment (sven1977) [inline rllib/tuned_examples/bc/cartpole_bc.py:55]: nice!

File: rllib/tuned_examples/bc/benchmark_rlunplugged_atari_pong_bc.py [hunk 3]
```diff
@@ -10,32 +10,30 @@
...
 from ray.rllib.core.columns import Columns
-from ray.rllib.core.rl_module.default_model_config import DefaultModelConfig
 from ray.rllib.env.wrappers.atari_wrappers import wrap_atari_for_new_api_stack
 from ray.rllib.utils.annotations import override
 from ray.rllib.utils.test_utils import (
     add_rllib_example_script_args,
 )
-from ray import tune
+
+# Register all ALE environments with gymnasium.
```
Comment (sven1977) [inline rllib/tuned_examples/bc/benchmark_rlunplugged_atari_pong_bc.py:32]: This is not needed for gymnasium 1.x.

File: rllib/tuned_examples/bc/benchmark_rlunplugged_atari_pong_bc.py [hunk 4]
```diff
@@ -10,32 +10,30 @@
       d_t: float
    }
 """
-
-# from google.cloud import storage
+import ale_py
```
Comment (sven1977) [inline rllib/tuned_examples/bc/benchmark_rlunplugged_atari_pong_bc.py:13]: I don't think we need to import this here anymore.

File: rllib/offline/offline_data.py [hunk 5]
```diff
@@ -146,8 +147,7 @@ def sample(
                 # Add constructor `kwargs` when using remote learners.
                 fn_constructor_kwargs.update(
                     {
-                        "learner": self.learner_handles,
-                        "locality_hints": self.locality_hints,
+                        "learner": None,
```
Comment (sven1977) [inline rllib/offline/offline_data.py:150]: Wait, what changed here?

**Review by sven1977 — COMMENTED** [review]
Looks good. Just a handful of questions ...

File: rllib/offline/offline_data.py (hunk 5 above)
Comment (simonsays1980) [reply rllib/offline/offline_data.py:150]: Locality hints are gone. The learner will go in a third PR blocked by the Ray Data w/ Ray Tune issue

File: rllib/offline/tests/test_offline_prelearner.py [hunk 6]
```diff
@@ -74,7 +74,7 @@ def test_offline_prelearner_buffer_class(self):
         algo = self.config.build()
         # Build the `OfflinePreLearner` and add the learner.
         oplr = OfflinePreLearner(
-            self.config,
+            config=self.config,
```
Comment (sven1977) [inline rllib/offline/tests/test_offline_prelearner.py:77]: nice!

File: rllib/offline/offline_prelearner.py [hunk 7]
```diff
@@ -84,9 +84,9 @@ class OfflinePreLearner:
     @OverrideToImplementCustomLogic_CallToSuperRecommended
     def __init__(
         self,
+        *,
```
Comment (sven1977) [inline rllib/offline/offline_prelearner.py:87]: thanks!

**Review by sven1977 — APPROVED** [review]
LGTM now. Thanks @simonsays1980 !

**Pull request merged**